src/librustc_mir/transform/mod.rs

```diff
@@ -280,6 +281,8 @@ fn run_optimization_passes<'tcx>(
             &remove_noop_landing_pads::RemoveNoopLandingPads,
             &cleanup_post_borrowck::CleanupNonCodegenStatements,
             &simplify::SimplifyCfg::new("early-opt"),
+            // Gather Unevaluated consts to evaluate them regardless of being optimized out
+            &uneval_const_set::UnevalConstSet,
```

Comment:
The previous passes can remove constants from unreachable blocks, right?

Reply:
You're right but at the same time I've removed that code :).